applications/StructuralMechanicsApplication/custom_elements/updated_lagrangian.cpp

```diff
@@ -135,13 +138,16 @@ void UpdatedLagrangian::FinalizeSolutionStep(const ProcessInfo& rCurrentProcessI
         // Compute element kinematics B, F, DN_DX ...
         this->CalculateKinematicVariables(this_kinematic_variables, point_number, this->GetIntegrationMethod());
 
+        // Setting the variables for the CL
+        SetConstitutiveVariables(this_kinematic_variables, this_constitutive_variables, Values, point_number, integration_points);
+
         // Call the constitutive law to update material variables
         mConstitutiveLawVector[point_number]->FinalizeMaterialResponse(Values, GetStressMeasure());
 
         mConstitutiveLawVector[point_number]->FinalizeSolutionStep(
         GetProperties(),
-        GetGeometry(),
-        row( GetGeometry().ShapeFunctionsValues(  ), point_number ),
+        r_geometry,
+        row( r_geometry.ShapeFunctionsValues(  ), point_number ),
         rCurrentProcessInfo
         );
```

Comment:
this method has been deprecated since forever
cannot you use the new one instead?

Reply:
the shape functions you mean?

Reply:
no, this version of FinalizeSolutionStep is deprecated:
https://github.com/KratosMultiphysics/Kratos/blob/1f8bfe43b97c69eaa781e6a6475af2de36506d96/kratos/includes/constitutive_law.h#L1028-L1032

you should use this one instead (which is already called two lines above in the element)
https://github.com/KratosMultiphysics/Kratos/blob/1f8bfe43b97c69eaa781e6a6475af2de36506d96/kratos/includes/constitutive_law.h#L1166

Reply:
I've removed the deprecated one since the other method is, as you point out, alredy being called. :)